services/src/main/java/com/linkedin/kmf/services/TopicRebalancer.java

```diff
@@ -0,0 +1,427 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.services;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import kafka.admin.AdminUtils;
+import kafka.admin.PreferredReplicaLeaderElectionCommand;
+import kafka.cluster.Broker;
+import kafka.common.TopicAndPartition;
+import kafka.utils.ZkUtils;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.security.JaasUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.kmf.common.Utils.ZK_CONNECTION_TIMEOUT_MS;
+import static com.linkedin.kmf.common.Utils.ZK_SESSION_TIMEOUT_MS;
+
+import scala.collection.Seq;
+
+
+/**
+ * Runs this periodically to rebalance the monitored topic across brokers and to reassign leaders to brokers so that the
+ * monitored topic is sampling all the brokers evenly.
+ */
+class TopicRebalancer implements Runnable {
```

Comment:
It seems that the TopicRebalancer is itself a periodic task that can run independently of ProduceServce. How about we decouple it from ProduceService and have a standalone TopicBalanceService so that user can specify it in kafka-monitor.properties when they start KafkaMonitor? This allows code to be simpler (by decoupling TopicRebalanceService from ProduceService) and config to be simpler (by removing produce.topic.rebalanceEnabled and put all rebalance related config under RebalanceServiceConfig)